### PR TITLE
mgr/vol: reuse code to spawn threads

### PR DESCRIPTION
**Pull request by rishabh-d-dave**
Same block of code is written twice to spawn threads in async_job.py.
This has already created an inconsistency in naming of threads. Have
a single method for this and use it instead.

This allows not allows reusing code but also reduces such inconsistency.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by rishabh-d-dave** [comment]
jenkins test make check

**Review by batrick — APPROVED** [review]

**Comment by rishabh-d-dave** [comment]
This PR is under test in [https://tracker.ceph.com/issues/68395](https://tracker.ceph.com/issues/68395).

**Comment by rishabh-d-dave** [comment]
This PR is under test in [https://tracker.ceph.com/issues/68419](https://tracker.ceph.com/issues/68419).

File: src/pybind/mgr/volumes/fs/async_job.py [hunk 1]
```diff
@@ -136,11 +136,31 @@ def __init__(self, volume_client, name_pfx, nr_concurrent_jobs):
...
         self.threads = []
-        for i in range(self.nr_concurrent_jobs):
-            self.threads.append(JobThread(self, volume_client, name="{0}.{1}".format(self.name_pfx, i)))
-            self.threads[-1].start()
+        self.spawn_all_threads()
         self.start()
 
+    def spawn_new_thread(self, suffix):
+        t_name = f'{self.name_pfx}.{time.time()}.{suffix}'
+        log.info(f'spawning new thread with name {t_name}')
```
Comment (vshankar) [inline src/pybind/mgr/volumes/fs/async_job.py:144]: why are these in info log level? They should be at debug level IMO.

**Review by vshankar — CHANGES_REQUESTED** [review]

File: src/pybind/mgr/volumes/fs/async_job.py (hunk 1 above)
Comment (rishabh-d-dave) [reply src/pybind/mgr/volumes/fs/async_job.py:144]: This one is pretty useful bit of information, it should be an info level IMO.

**Review by joscollin — COMMENTED** [review]
@rishabh-d-dave 
What's the motivation behind adding the helpers ? (just to understand). They are used only once.

File: src/pybind/mgr/volumes/fs/async_job.py (hunk 1 above)
Comment (vshankar) [reply src/pybind/mgr/volumes/fs/async_job.py:144]: It just tells it's spawning a thread. INFO should be used when you dump something like counters, etc. which helps in understanding how those values change over time.

**Comment by rishabh-d-dave** [comment]
> @rishabh-d-dave What's the motivation behind adding the helpers ? (just to understand). They are used only once.

Not exactly, both helper call the third helper. The idea is to have same code launch threads for the sake of uniformity. Earlier threads names were different.

**Comment by rishabh-d-dave** [comment]
Rebased to resolve the conflict.

**Review by rishabh-d-dave — COMMENTED** [review]
QA run was successful - https://tracker.ceph.com/projects/cephfs/wiki/Main#wip-rishabh-testing-20241007102024

File: src/pybind/mgr/volumes/fs/async_job.py (hunk 1 above)
Comment (rishabh-d-dave) [reply src/pybind/mgr/volumes/fs/async_job.py]: Done. PTAL.

**Review by vshankar — APPROVED** [review]
If info->debug was the only change and QA run was done, please go ahead and merge.

**Pull request merged**